app/js/enhancedDate.js

```diff
@@ -1,5 +1,133 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function (date) {
+// still maintains access to all globals
+  
+  var setDate = {};
+
+//Stores the passed-in date.  
+  setDate = function (date) {
+      var myDate = date;
+    
+//Do some type-checking to make sure you have a valid type.    
+      if(typeof myDate !== null)
+      {
+//Takes milliseconds after epoch or a date object. 
+        
+//Do some type-checking to make sure you have a valid type. 
+        if (typeof myDate == Date() || typeof myDate == "number"){
+          myDate = date;
+        } else {
+          myDate = false;
```

Comment:
Rather than setting myDate to false, just default to creating a new Date object without passing anything in to it: myDate  = new Date( );